python/docs/source/development/contributing.rst

```diff
@@ -155,18 +155,13 @@ Now, you can start developing and `running the tests <testing.rst>`_.
 Contributing and Maintaining Type Hints
 ----------------------------------------
 
-PySpark type hints are provided using stub files, placed in the same directory as the annotated module, with exception to:
-
-* ``# type: ignore`` in modules which don't have their own stubs (tests, examples and non-public API). 
-* pandas API on Spark (``pyspark.pandas`` package) where the type hints are inlined.
-
-As a rule of thumb, only public API is annotated.
+PySpark type hints are inlined, to take advantage of static type checking within the functions.
```

Comment:
I think we should keep "As a rule of thumb, only public API is annotated." I think it's not "within the functions" since it include parameters, etc.